### PR TITLE
[MM-30379] add MM-T433, MM-T437 and MM-T438

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Adds following test cases:
- [MM-T433](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T433)
- [MM-T437](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T437)
- [MM-T438](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T438)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30379

### CODE REVIEW

File: e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        cy.get(`#${teamA.name}TeamButton`, {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').click();
+
+        // * Verify there is no cross contamination between teams.
+        cy.get('#sidebarItem_town-square').should('not.have.class', 'unread-title');
+
+        // * DM Channel list should be the same on both teams with no missing names.
+        cy.get('#directChannelList').findByText(`${userB.username}`).should('be.visible');
+        cy.get('#directChannelList').findByText(`${userC.username}`).should('be.visible');
+
+        // * Channel viewed on team a before switching should be the one that displays after switching back (Town Square does not briefly show).
```
Comment (agnivade) [inline e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js:96]: on _a_ team.

Also - what does "Town square does not briefly show" mean?

**Review by agnivade — CHANGES_REQUESTED** [review]
Just needs some reorganization of promises. Looks good otherwise.

File: e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            userA = user;
+            teamA = team;
+            testChannel = channel;
+
+            cy.apiCreateUser().then(({user: otherUser}) => {
+                userB = otherUser;
+                return cy.apiAddUserToTeam(team.id, userB.id);
+            }).then(() => {
+                return cy.apiCreateUser().then(({user: otherUser}) => {
```
Comment (agnivade) [inline e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js:32]: You can chain this `then` outside too:

```js
cy.apiCreateUser().then(({user: otherUser}) => {
                userB = otherUser;
                return cy.apiAddUserToTeam(team.id, userB.id);
            }).then(() => {
                return cy.apiCreateUser();
            }).then(({user: otherUser}) => {
                    userC = otherUser;
                    return cy.apiAddUserToTeam(team.id, userC.id);
            }).then(() => {
                return cy.apiCreateTeam('team', 'Team');
            }).then(({team: otherTeam}) => {
                    teamB = otherTeam;
                    return cy.apiAddUserToTeam(otherTeam.id, userA.id);
            }).then(() => {
                    return cy.apiAddUserToTeam(otherTeam.id, userB.id);
            });
```

File: e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        cy.visit(`/${teamA.name}/channels/town-square`);
+    });
+
+    it('MM-T433 Switch teams', () => {
+        // # Open several DM channels, including accounts that are not on Team B.
+        cy.apiCreateDirectChannel([userA.id, userB.id]).then(() => {
+            cy.visit(`/${teamA.name}/channels/${userA.id}__${userB.id}`).wait(TIMEOUTS.FIVE_SEC);
+            cy.postMessage(':)');
+        }).then(() => {
+            cy.apiCreateDirectChannel([userA.id, userC.id]).then(() => {
```
Comment (agnivade) [inline e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js:60]: `cy.apiCreateDirectChannel` should be inside the parent then. Return it from there, and then place the inner callback after.

```js
 cy.apiCreateDirectChannel([userA.id, userB.id]).then(() => {
            cy.visit(`/${teamA.name}/channels/${userA.id}__${userB.id}`).wait(TIMEOUTS.FIVE_SEC);
            cy.postMessage(':)');
           return cy.apiCreateDirectChannel([userA.id, userC.id])
        }).then(() => {
                cy.visit(`/${teamA.name}/channels/${userA.id}__${userC.id}`).wait(TIMEOUTS.FIVE_SEC);
                cy.postMessage(':(');
        });
```

File: e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js (hunk 1 above)
Comment (isacikgoz) [reply e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js]: I believe it means that it should not open Town Square after we return back to Team A.

File: e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js (hunk 2 above)
Comment (isacikgoz) [reply e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js]: 😅 Thanks for clarifying 🙏

**Review by agnivade — APPROVED** [review]
:+1:

File: e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js (hunk 1 above)
Comment (agnivade) [reply e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js]: Oh it's a statement. I thought it was a comment about some quirk.

**Review by saturninoabril — APPROVED** [review]
Thanks @isacikgoz! Tested and passed. LGTM.

**Review by lindy65 — APPROVED** [review]
Thanks @isacikgoz - looks good to me 👍

**Pull request merged**